couchbase3/src/main/scala/org/apache/pekko/stream/connectors/couchbase3/scaladsl/CouchbaseSource.scala

```diff
@@ -0,0 +1,226 @@
+      .map(_.contentAs(target))
+
+  /**
+   * like Get[[CouchbaseSource.get]], batch get documents from collection by ScanType[[ScanType]]
+   */
+  def scan(scanType: ScanType, options: ScanOptions = ScanOptions.scanOptions())(
+      implicit asyncCollection: AsyncCollection): Source[ScanResult, NotUsed] =
+    Source
+      .completionStage(asyncCollection.scan(scanType, options))
+      .flatMapConcat(res => Source.fromJavaStream(() => res.stream()))
+
+  /**
+   * reference to [[CouchbaseSource.scan]], deserialize to Couchbase JsonObject
+   */
+  def scanJson(scanType: ScanType, options: ScanOptions = ScanOptions.scanOptions())(
+      implicit asyncCollection: AsyncCollection): Source[JsonObject, NotUsed] =
+    scan(scanType, options)
+      .map(_.contentAsObject())
+
+  /**
+   * reference to [[CouchbaseSource.scan]], deserialize to class
+   * If you add DefaultScalaModule to jackson of couchbase, it could deserialize to scala class
+   */
+  def scanObject[T](scanType: ScanType, target: Class[T],
+      options: ScanOptions = ScanOptions.scanOptions())(
+      implicit asyncCollection: AsyncCollection): Source[T, NotUsed] =
+    scan(scanType, options)
+      .map(_.contentAs(target))
+
+  /**
+   * reference to [[CouchbaseSource.scan]], [[CouchbaseSource.scanObject]], deserialize to class with Generics
+   */
+  def scanType[T](scanType: ScanType, target: TypeRef[T],
+      options: ScanOptions = ScanOptions.scanOptions())(
+      implicit asyncCollection: AsyncCollection): Source[T, NotUsed] =
+    scan(scanType, options)
+      .map(_.contentAs(target))
+
+  /**
+   * N1QL query in a Scope.
+   *
+   * QueryResult contains List<Row>, every Row is a json like
+   * <pre>
+   *   {
+   *     "collectionName": Document
+   *   }
+   * </pre>
+   */
+  def query(statement: String, options: QueryOptions = QueryOptions.queryOptions())(
+      implicit scope: AsyncScope): Source[QueryResult, NotUsed] = {
+    Source
+      .completionStage(scope.query(statement, options))
+  }
+
+  /**
+   * N1QL query in a Scope.
+   * @param statement N1QL query sql
+   * @see [[CouchbaseSource.query]]
+   */
+  def queryJson(
+      statement: String,
+      options: QueryOptions = QueryOptions.queryOptions())(
+      implicit scope: AsyncScope): Source[JsonObject, NotUsed] = {
+    query(statement, options)
+      .flatMapConcat { res =>
+        Source.fromJavaStream(() =>
+          res.rowsAsObject().stream.flatMap[JsonObject] { json =>
+            json.getNames.stream().map(collection => json.getObject(collection))
+          })
+      }
+  }
+
+  /**
+   * Performs an Analytics query , QueryResult contains List<Row>
+   * <p>
+   * val rows: List[JsonObject] = QueryResult.rowsAsObject, every Row is a json with CollectionName key
+   * <pre>
+   *    {
+   *      "collectionName": Document
+   *    }
+   * </pre>
+   * </p>
+   * <p>
+   *  Warning: couchbase-community not support analyticsQuery, we not test this API
+   * </p>
```

Comment:
can you provide a link that proves that this is unsupported? We may be better off removing these methods.

Reply:
I discovered it while writing unit tests that when I added it, the container could not start. I observed the logs in the Docker and it said it was not supported.
I have added a link

Reply:
query and analyticsQuery are very import api, because more people like to use sql. I don't want to drop this api. 
I like to use sql in my application.
We added a warning, I think it's enough, and there's no problem with the code,the api smilimar with query.